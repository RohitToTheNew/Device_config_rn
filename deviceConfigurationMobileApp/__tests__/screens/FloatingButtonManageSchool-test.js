// Mock the useBackButton hook from @react-navigation/native
jest.mock('@react-navigation/native/lib/commonjs/useBackButton.native', () => ({
  default: jest.fn(),
  __esModule: true,
}));

// Mock react-redux to provide showManageSchool state
jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn((selector) => {
      return selector({
        app: {
          showManageSchool: true,
        },
      });
    }),
    useDispatch: jest.fn().mockImplementation(() => jest.fn()),
  };
});

import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import FloatingButtonManageSchool from '../../src/components/floatingButtonManageSchool';

// Use fake timers to handle async operations
jest.useFakeTimers();

let tree;

const props = {
  extraStyles: undefined,
  navigation: { push: jest.fn() },
};

describe('on Floating button for Manage School render', () => {
  beforeEach(() => {
    let component;
    act(() => {
      component = renderer.create(
        <NavigationContainer>
          <FloatingButtonManageSchool {...props} />
        </NavigationContainer>
      );
    });
    tree = component.toJSON();
  });

  it('should render Floating button correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});

describe('Floating button icon', () => {
  it('if manageSchoolIcon pressed navigate to manageSchool', () => {
    let component;
    const navigationSpy = jest.spyOn(props.navigation, 'push');

    act(() => {
      component = renderer.create(
        <NavigationContainer>
          <FloatingButtonManageSchool {...props} />
        </NavigationContainer>,
      );
    });

    const manageSchoolIcon = component.root.findByProps({
      testID: 'manageSchoolButton',
    });

    act(() => {
      manageSchoolIcon.props.onPress();
    });

    expect(navigationSpy).toHaveBeenCalled();
  });
});
