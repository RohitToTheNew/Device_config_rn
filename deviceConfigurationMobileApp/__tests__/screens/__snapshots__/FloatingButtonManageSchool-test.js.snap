// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on Floating button for Manage School render should render Floating button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#5D9D52",
      "borderRadius": 8,
      "bottom": 60,
      "flexDirection": "row",
      "height": 108,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "width": 396,
    }
  }
  testID="manageSchoolButton"
>
  <
    height={48}
    width={48}
  />
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "CenturyGothic",
        "fontSize": 26,
        "fontWeight": "700",
        "lineHeight": 30,
        "marginStart": 27,
      }
    }
  >
    [missing "en.manageSchools" translation]
  </Text>
</View>
`;
